### DISCUSSION TITLE
Mongo CQRS support

### DISCUSSION THREAD

**Discussion by o-shevchenko** [micronaut-projects/micronaut-data#1503] [Q&A]
Is it possible to use different entities for reading and write with Micronaut Data Mongo `CrudRepository`?

Use case: Let's say we have two entities `Parent` and `Child`.
```
data class Parent(
    @field:Id
    @GeneratedValue
    var id: Id = "",
    val name: String
)

data class Child(
    @field:Id
    @GeneratedValue
    var id: Id = "",
    val name: String
    val parentId: Id
)
```
We want to be able to save Child using Child entity specified above, but read entity to `ChildQuery` to read data from Mongo by a single call from two entities.
```
data class Child(
    @field:Id
    @GeneratedValue
    var id: Id = "",
    val name: String
    @Relation(value = Relation.Kind.MANY_TO_ONE)
    val parent: Parent
)

or

data class Child(
    @field:Id
    @GeneratedValue
    var id: Id = "",
    val name: String
    val parentName: String
)
```
How it can be done using a single repository?
```
@MongoRepository
interface ChildRepository : CrudRepository<Child, Id>
```

**Comment by dstepanov** [comment]
It should be possible to define multiple entities that are referencing the same collection. Sorry for the late reply.

**Reply by o-shevchenko** [reply]
Any example of that?
I was able to create two repositories, one for read and another for write.

**Reply by dstepanov** [reply]
That's what I meant, two different entities and repositories
